pkg/eval/math/math_test.go

```diff
@@ -0,0 +1,104 @@
+package math
+
+import (
+	"fmt"
+	"math"
+	"testing"
+
+	"github.com/elves/elvish/pkg/eval"
+)
+
+var That = eval.That
+
+func TestMath(t *testing.T) {
+	setup := func(ev *eval.Evaler) { ev.Builtin.AddNs("math", Ns) }
+	eval.TestWithSetup(t, setup,
+		That(`put $math:pi`).Puts(math.Pi),
+		That(`put $math:e`).Puts(math.E),
+
+		That(`math:abs 2.1`).Puts(2.1),
+		That(`math:abs -2.1`).Puts(2.1),
+
+		That(`math:ceil 2.1`).Puts(3.0),
+		That(`math:ceil -2.1`).Puts(-2.0),
+
+		That(`math:floor 2.1`).Puts(2.0),
+		That(`math:floor -2.1`).Puts(-3.0),
+
+		That(`math:is-inf 1.3`).ThrowsAny(),
+		That(`math:is-inf 1.3 0`).Puts(false),
+		That(`math:is-inf inf 0`).Puts(true),
+		That(`math:is-inf inf 1`).Puts(true),
+		That(`math:is-inf -inf -1`).Puts(true),
+		That(`math:is-inf -inf 1`).Puts(false),
+		That(`math:is-inf -inf 0`).Puts(true),
+		That(`math:is-inf nan 0`).Puts(false),
+		That(`math:is-inf (float64 inf) 0`).Puts(true),
+		That(`math:is-inf (float64 inf) 1`).Puts(true),
+		That(`math:is-inf (float64 -inf) -1`).Puts(true),
+		That(`math:is-inf (float64 -inf) 1`).Puts(false),
+		That(`math:is-inf (float64 -inf) 0`).Puts(true),
+		That(`math:is-inf (float64 nan) 0`).Puts(false),
+
+		That(`math:is-nan 1.3`).Puts(false),
+		That(`math:is-nan inf`).Puts(false),
+		That(`math:is-nan nan`).Puts(true),
+		That(`math:is-nan (float64 inf)`).Puts(false),
+		That(`math:is-nan (float64 nan)`).Puts(true),
+
+		That(`math:round 2.1`).Puts(2.0),
+		That(`math:round -2.1`).Puts(-2.0),
+		That(`math:round 2.5`).Puts(3.0),
+		That(`math:round -2.5`).Puts(-3.0),
+		That(`math:round (float64 Inf)`).Puts(math.Inf(1)),
+		That(`math:round (float64 NaN)`).Puts(math.NaN()),
+
+		That(`math:round-to-even 2.1`).Puts(2.0),
+		That(`math:round-to-even -2.1`).Puts(-2.0),
+		That(`math:round-to-even 2.5`).Puts(2.0),
+		That(`math:round-to-even -2.5`).Puts(-2.0),
+		That(`math:round-to-even (float64 Inf)`).Puts(math.Inf(1)),
+		That(`math:round-to-even (float64 NaN)`).Puts(math.NaN()),
+
+		That(`math:trunc 2.1`).Puts(2.0),
+		That(`math:trunc -2.1`).Puts(-2.0),
+		That(`math:trunc 2.5`).Puts(2.0),
+		That(`math:trunc -2.5`).Puts(-2.0),
+		That(`math:trunc (float64 Inf)`).Puts(math.Inf(1)),
+		That(`math:trunc (float64 NaN)`).Puts(math.NaN()),
+
+		// That(`math:log 2.71828182845904523536028747135266249775724709369995`).Puts(1.0),
```

Comment:
Do not commit commented code; just remove it.

Reply:
Yes, that was an oversight from a prior experiment for improving the test. And the result of that experiment was also flawed since the test should be using the elvish `$math:e` constant, rather than interpolating the Go `math:E` constant as you pointed out in a different comment. :smile: